### PR TITLE
SAK-30211 added MailArchive entity provider

### PR DESCRIPTION
**Pull request by zqian**

### CODE REVIEW

File: mailarchive/mailarchive-tool/tool/pom.xml [hunk 1]
```diff
@@ -64,6 +64,18 @@
             <groupId>javax.servlet</groupId>
             <artifactId>servlet-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
```
Comment (jonespm) [inline mailarchive/mailarchive-tool/tool/pom.xml]: This is giving a build error in 11. All of these spring artifacts expanded out to their more specific dependencies in 10. (Like spring core)

https://confluence.sakaiproject.org/display/REL/Upgrading+a+tool+for+2.9+-+10.0+compatibility

File: mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java [hunk 2]
```diff
@@ -0,0 +1,822 @@
...
+	 * Prefix for this provider
+	 */
+	@Override
+	public String getEntityPrefix() {
+		return ENTITY_PREFIX;
+	}
+
+	// output formats
+	public String[] getHandledOutputFormats() {
+		return new String[] { Formats.JSON };
```
Comment (steveswinsburg) [inline mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java]: Should probably add Formats.XML to be consistent with the others

File: mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java [hunk 3]
```diff
@@ -0,0 +1,822 @@
...
+	}
+
+	// input formats
+	public String[] getHandledInputFormats() {
```
Comment (steveswinsburg) [inline mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java]: Not creating anything, this shouldn't be required. Might be implementing the wrong type or something too high up in the hierarchy.

File: mailarchive/mailarchive-tool/tool/pom.xml (hunk 1 above)
Comment (zqian) [inline mailarchive/mailarchive-tool/tool/pom.xml]: fixed.

File: mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java [hunk 4]
```diff
@@ -0,0 +1,827 @@
...
+		return ENTITY_PREFIX;
+	}
+
+	// output formats
+	public String[] getHandledOutputFormats() {
+		return new String[] { Formats.XML, Formats.JSON };
+	}
+
+	// input formats
+	public String[] getHandledInputFormats() {
```
Comment (zqian) [inline mailarchive/mailarchive-tool/tool/src/java/org/sakaiproject/mailarchive/entityprovider/MailArchiveEntityProvider.java]: getHandledInputFormats() is required by Inputable interface, and hence not removable.

**Pull request merged**